File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/playback/file/volume/preparation/volumemanagement/GivenATypicalMaxVolume/WhenChangingTheMaxVolume.java
package com.lasthopesoftware.bluewater.client.playback.file.volume.preparation.volumemanagement.GivenATypicalMaxVolume;


import com.lasthopesoftware.bluewater.client.playback.file.NoTransformVolumeManager;
import com.lasthopesoftware.bluewater.client.playback.file.volume.preparation.MaxFileVolumeManager;

import org.junit.BeforeClass;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class WhenChangingTheMaxVolume {
	private static NoTransformVolumeManager volumeManager;

	@BeforeClass
	public static void before() {
		volumeManager = new NoTransformVolumeManager();

		final MaxFileVolumeManager maxFileVolumeManager = new MaxFileVolumeManager(volumeManager);
		maxFileVolumeManager.setVolume(.58f);
		maxFileVolumeManager.setMaxFileVolume(.8f);
		maxFileVolumeManager.setMaxFileVolume(.47f);
	}

	@Test
	public void thenThePlaybackHandlerVolumeIsCorrectlySet() {
		assertThat(volumeManager.getVolume()).isEqualTo(.2726f);
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/playback/file/volume/preparation/volumemanagement/GivenATypicalMaxVolume/WhenChangingTheMaxVolume.kt
package com.lasthopesoftware.bluewater.client.playback.file.volume.preparation.volumemanagement.GivenATypicalMaxVolume

import com.lasthopesoftware.bluewater.client.playback.file.NoTransformVolumeManager
import com.lasthopesoftware.bluewater.client.playback.file.volume.preparation.MaxFileVolumeManager
import com.lasthopesoftware.bluewater.shared.promises.extensions.toExpiringFuture
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

class WhenChangingTheMaxVolume {

	private val volumeManager = NoTransformVolumeManager()

	@BeforeAll
	fun before() {
		val maxFileVolumeManager = MaxFileVolumeManager(volumeManager)
		maxFileVolumeManager.setVolume(.58f)
		maxFileVolumeManager.setMaxFileVolume(.8f)
		maxFileVolumeManager.setMaxFileVolume(.47f)
	}

	@Test
	fun `then the playback handler volume is correctly set`() {
		assertThat(volumeManager.volume.toExpiringFuture().get()).isEqualTo(.2726f)
	}
}
